python/mxnet/gluon/trainer.py

```diff
@@ -269,13 +273,13 @@ def step(self, batch_size, ignore_stale_grad=False):
             If true, ignores Parameters with stale gradient (gradient that has not
             been updated by `backward` after last step) and skip update.
         """
+        self._optimizer.rescale_grad = self._scale / batch_size
```

Comment:
If update_on_kvstore = True and distributed = True, the optimizer is set on the remote ps server. so if trainer.step is called twice with different `batchsize`, modifying self._optimizer.rescale_grad actually doesn't take any effect. We should throw a warning message if batchsize doesn't match when update_on_kvstore is True